cirq-web/cirq_web/widget.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from enum import Enum
+import uuid
+
+import cirq_web
+
+# Resolve the path so the bundle file can be accessed properly
+_DIST_PATH = Path(cirq_web.__file__).parents[1] / "cirq_ts" / "dist"
+
+
+class Env(Enum):
+    JUPYTER = 1
+    COLAB = 2
+    OTHER = 3
+
+
+def _to_script_tag(path: str) -> str:
+    """Dumps the contents of a particular bundle file into a script tag.
+
+    Args:
+        path: the path to the bundle file
+
+    Returns:
+        The bundle file as string (readable by browser) wrapped in HTML script tags.
+    """
+    bundle_file_path = path
+    bundle_file = open(bundle_file_path, 'r', encoding='utf-8')
+    bundle_file_contents = bundle_file.read()
+    bundle_file.close()
+    bundle_html = f'<script>{bundle_file_contents}</script>'
+
+    return bundle_html
+
+
+def write_output_file(output_directory: str, file_name: str, contents: str) -> Path:
```

Comment:
let's get rid of this method - it is very easy to write it in client code and really not typical to pull this out to a separate method: 

e.g you can write this in generate_html_file:
```
path_of_html_file = os.path.join(output_directory, file_name)
        with open(path_of_html_file) as f:
            f.write(contents)
```